core/src/test/java/io/parsingdata/metal/ReferenceValueExpressionSemanticsTest.java

```diff
@@ -56,14 +49,21 @@
             { "[2, 1, 2] b == a, c == b", sequenceMatchTransitive3, stream(2, 1, 2), enc(), false },
             { "[1, 2, 2] b == a, c == b", sequenceMatchTransitive3, stream(1, 2, 2), enc(), false },
             { "[1, 2, 3] b == a, c == b", sequenceMatchTransitive3, stream(1, 2, 3), enc(), false },
-            { "[1, 2, 1] a, a, first(a)", refList("a", "a", first("a")), stream(1, 2, 1), enc(), true },
-            { "[1, 2, 3] a, a, first(a)", refList("a", "a", first("a")), stream(1, 2, 3), enc(), false },
-            { "[1, 2, 3] a, a, first(b)", refList("a", "a", first("b")), stream(1, 2, 3), enc(), false },
+            { "[1, 2, 1] a, a, first(a)", refList("a", "a", first(ref("a"))), stream(1, 2, 1), enc(), true },
+            { "[1, 2, 3] a, a, first(a)", refList("a", "a", first(ref("a"))), stream(1, 2, 3), enc(), false },
+            { "[1, 2, 3] a, a, first(b)", refList("a", "a", first(ref("b"))), stream(1, 2, 3), enc(), false },
             { "[1, 2, 3] a, a, ref(b)", refList("a", "a", ref("b")), stream(1, 2, 3), enc(), false },
-            { "[1, 2, 0] a, b, offset(a)", refList("a", "b", offset("a")), stream(1, 2, 0), enc(), true },
-            { "[1, 2, 1] a, a, offset(a)", refList("a", "a", offset("a")), stream(1, 2, 1), enc(), true },
-            { "[1, 2, 2] a, b, offset(z)", refList("a", "b", offset("z")), stream(1, 2, 2), enc(), true },
-            { "[1, 2, 3] a, b, offset(c)", refList("a", "b", offset("c")), stream(1, 2, 3), enc(), false }
+            { "[1, 2, 3] a, a, last(a)", refList("a", "a", last(ref("a"))), stream(1, 2, 2), enc(), true },
```

Comment:
Test description does not match the stream data `[1, 2, 3]` vs `stream(1, 2, 2)`.